ops/ruby/devops-framework/devops.rb

```diff
@@ -0,0 +1,43 @@
+require 'logger'
+require 'optparse'
+require_relative 'tasks/count_monitoring_assets.rb'
+
+# Single entry point for the devops framework. This is the only true Ruby Script file. The
+# rest are classes.
+#
+# Based on input commands, this script delegates the work to  a task class, which should not need to
+# know anything about environment variables such as ARGV, the run directory,  etc. Tasks should also avoid
+# global variables like `logger`, so that we can control logging preferences from the top level.
+
+# TODO(jaycarlton): work out a scheme for task-specific options (i.e. subcommands)
+def parse_options
+  options = {}
+  OptionParser.new do |parser|
+    parser.on('-t', '--task [TASK]', String, 'Task to be in in each environment')
+    parser.on('-e', '--envs-file [ENVS]', String, 'Path to environments JSON file.')
+  end.parse!({into: options})
+
+  #Now raise an exception if we have not found a required arg
+  raise OptionParser::MissingArgument.new('task') if options[:task].nil?
```

Comment:
is there a way to do this generically, or error on any missing?

Reply:
There are a ton of ways to do it, but this looks like [the least bad](https://stackoverflow.com/a/2149183/12345554). I don't know why nobody has added that in 10+ years.